drivers/ruby/lib/net.rb

```diff
@@ -677,20 +677,40 @@ def init_socket
         context = create_context(@ssl_opts)
         @socket = OpenSSL::SSL::SSLSocket.new(@tcp_socket, context)
         @socket.sync_close = true
-        @socket.connect
+        socket_connect
         verify_cert!(@socket, context)
       else
         @socket = base_socket
+        socket_connect
       end
     end
 
     def base_socket
-      socket = TCPSocket.open(@host, @port)
+      @socket.close if @socket
```

Comment:
I don't think we should be calling `@socket.close` here.  This was previously a pure function, so it's a little weird to have it modifying a member variable.  If we want to be closing the socket, I think we should be doing it in `init_socket`.  In this case, though, `init_socket` is only called by `connect`, which checks that `@socket` isn't set immediately before calling it on line 663, so we should probably just omit this conditional close entirely.